indico/modules/rb/client/js/common/rooms/edit/RoomEditOptions.jsx

```diff
@@ -0,0 +1,162 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import {Dropdown, Form, Header, Tab} from 'semantic-ui-react';
+import React, {useCallback, useMemo} from 'react';
+import _ from 'lodash';
+import {FieldArray} from 'react-final-form-arrays';
+import PropTypes from 'prop-types';
+import shortid from 'shortid';
+import {Translate} from 'indico/react/i18n';
+import {FinalCheckbox, FinalField, FinalInput} from 'indico/react/forms';
+import EquipmentList from './EquipmentList';
+import DailyAvailability from './DailyAvailability';
+import NonBookablePeriods from './NonBookablePeriods';
+
+function RoomEditOptions({active, showEquipment, globalAttributes}) {
+  const attributeTitles = useMemo(
+    () => (globalAttributes ? _.fromPairs(globalAttributes.map(x => [x.name, x.title])) : []),
+    [globalAttributes]
+  );
+  const withKeyAttribute = useCallback(
```

Comment:
Remove this method since the methods below already generate the keys internally.